File: api/src/main/java/com/api/app/emum/MEM001.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum MEM001 implements CommonCode {
    ACTIVE("001", "정상회원", 1, "", ""),
    WITHDRAWN("002", "탈퇴회원", 2, "", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static MEM001 findByCode(String code) {
        return CommonCodeUtil.findByCode(MEM001.class, code);
    }
}

File: api/src/main/kotlin/com/api/app/emum/MEM001.kt
package com.api.app.emum

/** 회원상태코드: memberStatusCode */
enum class MEM001(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    ACTIVE("001", "정상회원", 1, "", ""),
    WITHDRAWN("002", "탈퇴회원", 2, "", ""),
    ;

    companion object {
        fun findByCode(code: String): MEM001? = CommonCodeUtil.findByCode(MEM001::class.java, code)
    }
}
